messaging/src/main/java/org/axonframework/deadline/dbscheduler/DbSchedulerDeadlineManager.java

```diff
@@ -235,7 +236,14 @@ public static Task<DbSchedulerHumanReadableDeadlineDetails> humanReadableTask(
     public void cancelSchedule(@Nonnull String deadlineName, @Nonnull String scheduleId) {
         Span span = spanFactory.createCancelScheduleSpan(deadlineName, scheduleId);
         runOnPrepareCommitOrNow(span.wrapRunnable(
-                () -> scheduler.cancel(new DbSchedulerDeadlineToken(scheduleId)))
+                () -> {
+                    try {
+                        scheduler.cancel(new DbSchedulerDeadlineToken(scheduleId));
+                    } catch (TaskInstanceNotFoundException e) {
+                        // handle gracefully
+                        logger.debug("Attempted to cancel task [{}] which does not exist anymore", scheduleId);
```

Comment:
Bit of a detail, but I don't think we can be a 100% sure it used to exist in the first place. 
Users may accidentally provide a `scheduledId` that's wrong, for example.

So, I'd do something like this:
```suggestion
                        logger.debug("Attempted to cancel task [{}] which does not exist. The task may already have been canceled or the given schedule identifier is incorrect.", scheduleId);
```